matminer/featurizers/dos.py

```diff
@@ -6,6 +6,78 @@
 from pymatgen.electronic_structure.dos import CompleteDos, FermiDos
 
 
+class SiteDOS(BaseFeaturizer):
+    """
+    report the fractional s/p/d/f dos for a particular site. a CompleteDos
+    object is required because knowledge of the structure is needed. this
+    featurizer will work for metals as well as semiconductors. if the dos is a
+    semiconductor, cbm and vbm will correspond to the two respective band
+    edges. if the dos is a metal, then cbm and vbm correspond to above and
+    below the fermi level, respectively.
+
+    Args:
+        decay_length (float in eV):
+            the dos is sampled by an exponential decay function. this parameter
+            sets the decay length of the exponential. three times the
+            decay_length corresponds to 10% sampling strength. there is a hard
+            cutoff at five times the decay length (1% sampling strength)
+        sampling_resolution (int):
+            number of points to sample dos
+        gaussian_smear (float in eV):
+            Gaussian smearing (sigma) around each sampled point in dos
+
+    Returns (list of floats):
+        cbm_score_i (float): fractional score for i in {s,p,d,f}
+        cbm_score_total (float): the total sum of all the {s,p,d,f} scores
+            this is usefull information when comparing the relative
+            contributions from multiples sites
+        vbm_score_i (float): fractional score for i in {s,p,d,f}
+        vbm_score_total (float): the total sum of all the {s,p,d,f} scores
+            this is usefull information when comparing the relative
+            contributions from multiples sites
+    """
+    def __init__(self, decay_length=0.1, sampling_resolution=100,
+                 gaussian_smear=0.05):
+        self.decay_length = decay_length
+        self.sampling_resolution = sampling_resolution
+        self.gaussian_smear = gaussian_smear
+
+    def featurize(self, dos, idx):
+        """
+        get dos scores for given site index
+
+        Args:
+            dos (pymatgen CompleteDos or their dict):
+                dos to featurize, must contain pdos and structure
+            idx (int): index of target site in structure.
+        """
+        if isinstance(dos, dict):
+            dos = CompleteDos.from_dict(dos)
+        if dos.structure is None:
+            raise ValueError('The input dos must contain the structure.')
+
+        orbscores = get_site_dos_scores(dos, self.decay_length,
```

Comment:
would be better if the order of parameters was:
- dos
- idx
- decay length
- sampling_resolution
- gaussian smear

It just makes more sense to me since the dos and idx are the most "important and required" parameters, the other ones are just adjustments that would sometimes even be optional params with defaults

Reply:
changed argument order